model/src/test_manager/manager_impl.rs

```diff
@@ -214,19 +214,42 @@ impl TestManager {
             ResourceState::Creation => "creation",
             ResourceState::Destruction => "destruction",
         };
-        pod_api
-            .list(&ListParams {
-                label_selector: Some(format!("job-name={}-{}", resource.into(), suffix)),
-                ..Default::default()
-            })
-            .await
-            .context(error::KubeSnafu { action: "get pods" })?
-            .items
-            .first()
-            .context(error::NotFoundSnafu {
-                what: "pod for test",
-            })
-            .map(|pod| pod.clone())
+        let mut resource_name = resource.into();
+        let resource_api = Api::<Resource>::namespaced(self.k8s_client.clone(), NAMESPACE);
+        let resource_crd = resource_api.get(&resource_name).await;
+        let result = match resource_crd {
+            // if the resource exists, retrieve the pod based on the truncated resource name + UID + resource state
+            Ok(resource_object) => {
+                resource_name.truncate(TRUNC_LEN);
+                return pod_api
+                    .list(&ListParams {
+                        label_selector: Some(format!(
+                            "job-name={}{}-{}",
+                            resource_name,
+                            resource_object
+                                .metadata
+                                .uid
+                                .as_ref()
+                                .unwrap_or(&"".to_string()),
+                            suffix
+                        )),
+                        ..Default::default()
```

Comment:
Let's convert this to a job name function (resource struct or resource client). (Also change the truncate in previous pr).

Reply:
When you say to change the truncate in previous pr, does that only refer to the truncate in `does_resource_destruction_pod_exist` from `cluster.rs`? The other one is in `new` in `context.rs` but to me it doesn't make sense to remove it there because the new `get_job_name` function will retrieve the resource CRD from the API and `new` already has the resource CRD.

Reply:
It would be nice to have all of the truncate logic in a single function that is called whenever the resource's job/pod name is needed. Probably belongs in `resource.rs`.

Reply:
Unless I'm misunderstanding something, I think the places where the job names need to be generated don't have access to the Resource struct. For example, when the command `cli logs --resource dup-1 --state creation` is executed, the TestManager has no idea where the resource CRD for that resource is, so the first API call will be needed to retrieve the UID, and the second API call to retrieve the actual creation pod.
It definitely seems reasonable to add a function to `resource_client.rs` and have the truncation logic consolidated there, but I don't see a way around two API calls.

Reply:
You would get the resource first.
`let resource = resource_client.get(my_resource)`
Then get the pod using `format!("job-name={}", resource.job_name(ResourceState::Create)`

Reply:
I addressed these changes in the new commit. Truncation now only happens in the `job_name` method in `resource.rs`.